addons/web/static/src/core/assets.js

```diff
@@ -65,125 +70,159 @@ export const loadCSS = memoize(function loadCSS(url) {
+ *      sequentially (so use this if the order matters) while the assets in
+ *      inner lists are loaded in parallel (use this for efficiency but only
+ *      if the order does not matter, should rarely be the case for assets).
+ * @param {string[]} [libs.cssLibs=[]]
+ *      The list of CSS files to load. They will all be loaded in parallel but
+ *      put in the DOM in the given order (only the order in the DOM is used
+ *      to determine priority of CSS rules, not loaded time).
+ * @param {Array<string|string[]>} [libs.jsLibs=[]]
+ *      The list of JS files to load. Each list item may be a string (the URL
+ *      of the file to load) or a list of strings. The first level is loaded
+ *      sequentially (so use this if the order matters) while the files in inner
+ *      lists are loaded in parallel (use this for efficiency but only
+ *      if the order does not matter).
+ * @param {string[]} [libs.cssContents=[]]
+ *      List of inline styles to add after loading the CSS files.
+ * @param {string[]} [libs.jsContents=[]]
+ *      List of inline scripts to add after loading the JS files.
+ *
+ * @returns {Promise}
  */
-export class LazyComponent extends Component {
-    setup() {
-        onWillStart(async () => {
-            await loadBundle(this.props.bundle);
-            this.Component = registry.category("lazy_components").get(this.props.Component);
+assets.loadLibs = async function loadLibs(libs) {
+    var defs = [];
+    var cssLibs = [libs.cssLibs || []];  // Force loading in parallel
+    defs.push(_loadArray(cssLibs, assets.loadCSS).then(function () {
+        if (libs.cssContents && libs.cssContents.length) {
+            const style = document.createElement("style");
+            style.innerHTML = libs.cssContents.join('\n');
+            document.head.appendChild(style);
+        }
+    }));
+    defs.push(_loadArray(libs.jsLibs || [], assets.loadJS).then(function () {
+        if (libs.jsContents && libs.jsContents.length) {
+            const script = document.createElement("script");
+            script.type = "text/javascript";
+            script.innerHTML = libs.jsContents.join('\n');
+            document.head.appendChild(script);
+        }
+    }));
+    await Promise.all(defs);
+    if (libs.assetLibs) {
+        await _loadArray(libs.assetLibs, async xmlID => {
+            const libs = await assets.getLibs(xmlID);
+            return assets.loadLibs(libs);
         });
     }
-}
-LazyComponent.template = xml`<t t-component="Component" t-props="props.props"/>`;
-LazyComponent.props = {
-    Component: String,
-    bundle: String,
-    props: { type: Object, optional: true },
+
+    async function _loadArray(array, loadCallback) {
+        for (const urlData of array) {
+            if (typeof urlData === 'string') {
+                await loadCallback(urlData);
+            } else {
+                await Promise.all(urlData.map(loadCallback));
+            }
+        }
+    }
```

Comment:
I don't think she should be outside. This is a utility only for the current method which is an implementation detail to reduce the number of rows. If we release it it will be part of the API.